aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/OutboxMutationFailedEvent.java

```diff
@@ -111,8 +96,83 @@ public M getModel() {
         return model;
     }
 
+    /**
+     * Gets the name of the model.
+     * @return the model name
+     */
+    @NonNull
+    public String getModelName() {
+        return model.getClass().getSimpleName();
+    }
+
     @Override
     public HubEvent<OutboxMutationFailedEvent<M>> toHubEvent() {
         return HubEvent.create(DataStoreChannelEventName.OUTBOX_MUTATION_FAILED, this);
     }
+
+    @Override
+    public String toString() {
+        return "OutboxMutationFailedEvent{" +
+                "errorType=" + errorType +
+                ", operation=" + operation +
+                ", model=" + model +
+                ", modelName=" + getModelName() +
+                '}';
+    }
+
+    /**
+     * Categorization of error types that caused a mutation publication to fail.
+     */
+    public enum MutationErrorType {
+        /**
+         * The mutation operation is not authorized for the user.
+         */
+        UNAUTHORIZED("Unauthorized"),
+
+        /**
+         * Fallback type for any error that is yet to be categorized.
+         */
+        UNKNOWN("Unknown");
+
+        private final String value;
+
+        MutationErrorType(String value) {
+            this.value = value;
+        }
+
+        /**
+         * Get the value of error type received from cloud.
+         * @return value of error type received from cloud
+         */
+        @NonNull
+        public String getValue() {
+            return value;
+        }
+
+        /**
+         * Returns an enum value of matching error type value.
+         * @param value the type of error that caused mutation publication failure
+         * @return An enum value of matching error type. If there is no match,
+         *          then return {@link MutationErrorType#UNKNOWN}.
+         */
+        @NonNull
+        public static MutationErrorType fromErrorType(@Nullable String value) {
+            try {
+                return MutationErrorType.valueOf(value);
```

Comment:
Hm, does `valueOf` check the `name()` (an uppercase String)? How would it know to inspect the `value` property, which is Pascal-cased? 

[One reference](https://www.tutorialspoint.com/java/lang/enum_valueof.htm) states:

> The java.lang.Enum.valueOf() method returns the enum constant of the specified enum type with the specified name. The name must match exactly an identifier used to declare an enum constant in this type.

Reply:
nice catch! will fix.